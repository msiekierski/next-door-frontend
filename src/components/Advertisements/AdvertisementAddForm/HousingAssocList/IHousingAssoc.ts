export interface IHousingAssoc {
  idAssoc: number;
  city: string;
  street: string;
  postalCode: string;
  name: string;
}
