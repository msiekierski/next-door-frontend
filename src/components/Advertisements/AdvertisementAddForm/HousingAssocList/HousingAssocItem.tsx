import React, { ChangeEvent, FunctionComponent } from "react";
import { Container, Form, ListGroup } from "react-bootstrap";
import { IHousingAssoc } from "./IHousingAssoc";
import { ADVERT_BOTTOM_PAYMENT } from "../../../../constants/constants";

type Props = IHousingAssoc & {
  setChecked: Function;
  setCharge: Function;
  setIsValid: Function;
};

const HousingAssocItem: FunctionComponent<Props> = ({ name, street, setChecked, idAssoc, setCharge, setIsValid }) => {
  function handleChecked(e: ChangeEvent<HTMLInputElement>) {
    if (e.target.checked) {
      setChecked((checked: number[]) => {
        const newChecked: number[] = [...checked, idAssoc];
        setCharge(100 * newChecked.length);
        if (newChecked.length >= 1) {
          setIsValid(true);
        }
        return newChecked;
      });
    } else {
      setChecked((checked: number[]) => {
        const newChecked: number[] = checked.filter((e) => e != idAssoc);
        setCharge(100 * newChecked.length);
        if (newChecked.length == 0) {
          setIsValid(false);
        }
        return newChecked;
      });
    }
  }

  return (
    <ListGroup.Item className={"pl-0 pr-0"}>
      <Container className={"d-flex justify-content-between"}>
        <Form.Label className={`mb-0`}>
          {name}, {street}
        </Form.Label>
        <Form.Check type="checkbox" onChange={handleChecked} />
      </Container>
    </ListGroup.Item>
  );
};

export default HousingAssocItem;
