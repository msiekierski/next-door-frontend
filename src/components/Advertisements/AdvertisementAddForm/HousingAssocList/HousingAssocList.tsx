import React, { FunctionComponent, useEffect, useState } from "react";
import HousingAssocItem from "./HousingAssocItem";
import { ListGroup } from "react-bootstrap";
import { getAllHousingAssoc } from "../../../../API/advertisement";
import { IHousingAssoc } from "./IHousingAssoc";

interface OwnProps {
  setChecked: Function;
  setCharge: Function;
  setIsValid: Function;
}

type Props = OwnProps;

const HousingAssocList: FunctionComponent<Props> = ({ setChecked, setCharge, setIsValid }) => {
  const [housingAssoc, setHousingAssoc] = useState<Array<IHousingAssoc>>([]);

  useEffect(() => {
    const fetchHousingAssoc = async () => setHousingAssoc(await getAllHousingAssoc());
    fetchHousingAssoc();
  }, []);

  return (
    <ListGroup variant={"flush"}>
      {housingAssoc.length
        ? housingAssoc.map((e, index) => (
            <HousingAssocItem
              key={index}
              idAssoc={e.idAssoc}
              city={e.city}
              street={e.street}
              postalCode={e.postalCode}
              name={e.name}
              setChecked={setChecked}
              setCharge={setCharge}
              setIsValid={setIsValid}
            />
          ))
        : "Loading"}
    </ListGroup>
  );
};

export default HousingAssocList;
