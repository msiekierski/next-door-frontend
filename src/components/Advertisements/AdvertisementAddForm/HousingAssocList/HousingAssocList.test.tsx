import React from "react";
import renderer from "react-test-renderer";
import HousingAssocList from "./HousingAssocList"

interface Props {
    setCharge: Function;
}

const props: Props = {
    setCharge: () => {}
}


it("renders correctly", () => {
  const tree = renderer.create(<HousingAssocList {...props} />);
  expect(tree).toMatchSnapshot();
});
