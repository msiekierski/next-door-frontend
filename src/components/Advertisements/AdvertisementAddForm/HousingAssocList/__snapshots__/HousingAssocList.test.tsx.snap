// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="list-group list-group-flush"
  onKeyDown={[Function]}
>
  Loading
</div>
`;
